tests/experimental/datasources/test_viral_snippets.py

```diff
@@ -0,0 +1,157 @@
+                            {"metadata_key": "month"},
+                        ],
+                    },
+                },
+            }
+        }
+    }
+
+
+@pytest.fixture
+def zep_only_config(zep_config_dict: dict) -> GxConfig:
+    zep_config = GxConfig.parse_obj(zep_config_dict)
+    assert zep_config.datasources
+    return zep_config
+
+
+@pytest.fixture
+def file_dc_config_dir_init(tmp_path: pathlib.Path) -> pathlib.Path:
+    """
+    Initialize an regular/old-style FileDataContext project config directory.
+    Removed on teardown.
+    """
+    gx_yml = tmp_path / FileDataContext.GX_DIR / FileDataContext.GX_YML
+    assert gx_yml.exists() is False
+    FileDataContext.create(tmp_path)
+    assert gx_yml.exists()
+
+    tmp_gx_dir = gx_yml.parent.absolute()
+    LOGGER.info(f"tmp_gx_dir -> {tmp_gx_dir}")
+    return tmp_gx_dir
+
+
+@pytest.fixture
+def zep_yaml_config_file(
+    file_dc_config_dir_init: pathlib.Path, zep_only_config: GxConfig
+) -> pathlib.Path:
+    """
+    Dump the provided GxConfig to a temporary path. File is removed during test teardown.
+    """
+    config_file_path = file_dc_config_dir_init / FileDataContext.ZEP_YAML
+
+    assert config_file_path.exists() is False
+
+    zep_only_config.yaml(config_file_path)
+    assert config_file_path.exists() is True
+    return config_file_path
+
+
+def test_load_an_existing_config(
+    zep_yaml_config_file: pathlib.Path, zep_only_config: GxConfig
+):
+    context = get_context(
+        context_root_dir=zep_yaml_config_file.parent, cloud_mode=False
+    )
+
+    assert context.zep_config == zep_only_config
+
+
+@pytest.fixture
+@functools.lru_cache(maxsize=1)
+def zep_file_context(zep_yaml_config_file: pathlib.Path) -> FileDataContext:
+    context = get_context(
+        context_root_dir=zep_yaml_config_file.parent, cloud_mode=False
+    )
+    assert isinstance(context, FileDataContext)
+    return context
+
+
+def test_serialize_zep_config(zep_file_context: FileDataContext):
+    dumped_yaml: str = zep_file_context.zep_config.yaml()
+    print(f"  Dumped Config\n\n{dumped_yaml}\n")
+
+    assert zep_file_context.zep_config.datasources
+
+    for ds_name, datasource in zep_file_context.zep_config.datasources.items():
+        assert ds_name in dumped_yaml
+
+        for asset_name in datasource.assets.keys():
+            assert asset_name in dumped_yaml
+
+
+def test_zep_simple_validate_workflow(zep_file_context: FileDataContext):
+    my_datasource: Datasource = zep_file_context.get_datasource("my_sql_ds")
+    my_asset: DataAsset = my_datasource.get_asset("my_asset")
+
+    batch_request = my_asset.get_batch_request({"year": 2019, "month": 1})
+    validator = zep_file_context.get_validator(batch_request=batch_request)
+    result = validator.expect_column_max_to_be_between(
+        column="passenger_count", min_value=1, max_value=12
+    )
+    print(f"  results ->\n{pf(result)}")
+    assert result["success"] == True
```

Comment:
I tried to keep this snippet as simple as possible.
Please LMK if this could be simplified further, or conversely if there's something I'm missing.

Reply:
This looks good to me. You could combine the first 3 lines if you wanted, not sure if it's simpler or not:
```
batch_request = zep_file_context.get_datasource("my_sql_ds").get_asset("my_asset").get_batch_request({"year": 2019, "month": 1})
```

Reply:
That's much better